crypto/merkletrie/cache.go

```diff
@@ -118,6 +118,34 @@ func (mtc *merkleTrieCache) allocateNewNode() (pnode *node, nid storedNodeIdenti
 	return newNode, nextID
 }
 
+// refurbishNode releases a given node and reallocate a new node while avoiding changing the underlaying buffer.
+func (mtc *merkleTrieCache) refurbishNode(nid storedNodeIdentifier) (nextID storedNodeIdentifier) {
```

Comment:
The goal was to avoid the reallocation of the `&node` object, since it was identified by the profiler to be one of the most allocated objects. Go takes a long time before it starts releasing memory, so it helps to "recycle" recently used objects.